blog/2025-02-26-blockmax-wand/index.mdx

```diff
@@ -0,0 +1,228 @@
+
+Keyword search works by comparing the terms in your queries to the terms in your database documents, giving higher scores to rarer terms and terms that show up more often in your documents.
+
+Keyword search requires us to first define what words we want to search. A big part of this process is [tokenization](/developers/academy/py/tokenization/basics). It splits the input documents into [tokens](https://en.wikipedia.org/wiki/Lexical_analysis#Token) (i.e. terms, numbers or anything else one would consider important to be searched individually).  
+In this simple example, consider a dataset made of three documents with a single property, title, composed of a single sentence and a *whitespace* tokenizer that lowercases the terms and splits them by whitespace. 
+
+| Doc Id | Document Title | Tokenizated Title |
+| :---- | :---- | :---- |
+| 1 | A Web Developers Guide to Hybrid Search | \[“a”, “web”, “developers”, “guide”, “to”, “hybrid”, “search”\] |
+| 2 | Unlocking the Power of Hybrid Search | \[“unlocking”, “the”, “power”, “of”, “hybrid”, “search”\] |
+| 3 | Vector Library versus Vector Database | \[“vector”, “library”, “versus”, “vector”, “database”\] |
+
+**Table**: Example dataset with tokenized titles.
+
+Now we have turned documents into a [bag-of-words](https://en.wikipedia.org/wiki/Bag-of-words_model) model, where we can find for individual query terms in the sentences. But having to go through all documents to find the ones that have query terms isn’t effective.   
+
+This is where the *inverted* part of the [inverted index](https://en.wikipedia.org/wiki/Inverted_index) comes from: instead of going document \-\> term, create an index that does from term \-\> documents. It works like the indexes at the end of books, but instead of mapping terms to book pages, we map terms to documents using posting lists.
+
+A posting list is a list of "postings", which contain the information needed to score documents: doc id (to identify which documents have the term), [term frequency (TF)](https://en.wikipedia.org/wiki/Tf%E2%80%93idf), which represents the number of times the term is part of the property. For example, TF(“vector”) for doc 3 is 2, as it shows up twice.
+
+| Term | Posting List |
+| :---- | :---- |
+| hybrid | (Doc 1, TF: 1); (Doc 2, TF 1\) |
+| search | (Doc 1, TF: 1); (Doc 2, TF 1\) |
+| vector | (Doc 3, TF: 2\) |
+
+**Table**: Posting lists for terms hybrid, search and vector in the example dataset.
+
+When a user searches for a query, we tokenize the query in the same way we tokenized the documents.
+Thus, if you want to search for `“Hybrid search or Vector search”`, we get the terms `[“hybrid”, “search”, “or”, “vector”]`.
+Inspecting the posting lists for each term, we can see that documents 1, 2 and 3 have at least one of the query terms.
+But we still need to score them to see which ones are the most relevant to the query.
+
+## TF-IDF and BM25
+
+In keyword search, not all terms are equally important.   
```

Comment:
Wdyt about:

Keyword search is based on the intuition that some words, or terms, are more important than others.